src/utils/I18nManager.js

```diff
@@ -226,11 +232,19 @@ class I18nManager {
    *  try to use plain object for all messages without nesting.
    */
   getMessage = (path, args = {}) => {
-    const locales = generateFallbackLocaleList(this.locale, this.fallbackLocale);
+    let message;
 
-    let message = undefined;
-    for (let localeIndex = 0; localeIndex < locales.length && message === undefined; localeIndex++) {
-      message = lodash.get(this.localeBundles[locales[localeIndex]], path);
+    if (typeof this.overwriteLocaleMessages === 'function') {
```

Comment:
I would put this validation into the constructor and throw an exception in case if it is not true.  By default, I would generate the function that returns an empty object (not messages), and again in the constructor.